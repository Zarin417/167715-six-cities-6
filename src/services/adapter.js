export const adaptOffersData = (data) => {
  const adaptedData = ({
    ...data,
    isFavorite: data.is_favorite,
    isPremium: data.is_premium,
    previewImage: data.preview_image,
    maxAdults: data.max_adults,
    host: {
      ...data.host,
      avatarUrl: data.host.avatar_url,
      isPro: data.host.is_pro
    }
  });

  delete adaptedData.is_favorite;
  delete adaptedData.is_premium;
  delete adaptedData.preview_image;
  delete adaptedData.max_adults;
  delete adaptedData.host.avatar_url;
  delete adaptedData.host.is_pro;

  return adaptedData;
};

export const adaptReviewsData = (data) => {
  const adaptedData = {
    ...data,
    user: {
      ...data.user,
      avatarUrl: data.user.avatar_url,
      isPro: data.user.is_pro
    }
  };

  delete adaptedData.user.avatar_url;
  delete adaptedData.user.is_pro;

  return adaptedData;
};
