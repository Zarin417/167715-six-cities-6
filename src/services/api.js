import axios from "axios";

const TIMEOUT = 5000;
const BASE_URL = `https://6.react.pages.academy/six-cities`;

const statusCode = {
  UNAUTHORIZED: 401,
};

export const createAPI = (unAuthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    timeout: TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onError = (error) => {
    const {response} = error;

    if (response.status === statusCode.UNAUTHORIZED) {
      unAuthorized();
    }

    throw error;
  };

  axios.interceptors.response.use(onSuccess, onError);

  return api;
};
