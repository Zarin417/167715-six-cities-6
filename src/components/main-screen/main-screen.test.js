import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {createMemoryHistory} from "history";
import {Provider} from "react-redux";
import MainScreen from "./main-screen";
import {Router} from "react-router";
import thunk from "redux-thunk";
import {createAPI} from "../../services/api";
import TestWrapper from "../../services/test-wrapper/test-wrapper";

const api = createAPI(() => {});
const mockStore = configureStore([thunk.withExtraArgument(api)]);
let history;

describe(`Test 'MainScreen'`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });

  it(`MainScreen should render 'Loader' when data is loading`, () => {
    const store = mockStore({
      OFFER: {isOffersLoaded: false, offers: []}
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <MainScreen />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`MainScreen should render empty page if there are no offers`, () => {
    const store = mockStore({
      OFFER: {isOffersLoaded: true, offers: []},
      USER: {isLoggedIn: false},
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <MainScreen />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`MainScreen should render correctly`, () => {
    const {container} = render(
        <TestWrapper>
          <MainScreen />
        </TestWrapper>
    );

    expect(container).toMatchSnapshot();
  });
});
