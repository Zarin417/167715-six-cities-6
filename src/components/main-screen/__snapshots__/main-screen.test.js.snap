// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'MainScreen' MainScreen should render 'Loader' when data is loading 1`] = `
<div>
  <div
    class="loader"
  >
    <div
      class="loader__dual-ring"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`Test 'MainScreen' MainScreen should render correctly 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
                style="display: flex;"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    keks@test.com
                  </span>
                </a>
                <button
                  style="margin-left: 10px;"
                >
                  Log Out
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index false"
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
                href="/city/Paris"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Cologne"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Brussels"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Amsterdam"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Hamburg"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Dusseldorf"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container container"
        >
          <section
            class="cities__places places"
          >
            <h2
              class="visually-hidden"
            >
              Places
            </h2>
            <b
              class="places__found"
            >
              1
              place
            to stay in Paris
            </b>
            <form
              action="#"
              class="places__sorting"
              method="get"
            >
              <span
                class="places__sorting-caption"
              >
                Sort by 
              </span>
              <span
                class="places__sorting-type"
                tabindex="0"
              >
                Popular
                <svg
                  class="places__sorting-arrow"
                  height="4"
                  width="7"
                >
                  <use
                    xlink:href="#icon-arrow-select"
                  />
                </svg>
              </span>
              <ul
                class="places__options places__options--custom false"
              >
                <li
                  class="places__option places__option--active"
                  tabindex="0"
                  type="popular"
                >
                  Popular
                </li>
                <li
                  class="places__option false"
                  tabindex="0"
                  type="price: low to high"
                >
                  Price: low to high
                </li>
                <li
                  class="places__option false"
                  tabindex="0"
                  type="price: high to low"
                >
                  Price: high to low
                </li>
                <li
                  class="places__option false"
                  tabindex="0"
                  type="top rated first"
                >
                  Top rated first
                </li>
              </ul>
            </form>
            <div
              class="cities__places-list tabs__content places__list"
            >
              <article
                class="cities__place-card place-card"
              >
                <div
                  class="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  class="cities__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/11"
                  >
                    <img
                      alt="Place image"
                      class="place-card__image"
                      height="200"
                      src=""
                      width="260"
                    />
                  </a>
                </div>
                <div
                  class=" place-card__info"
                >
                  <div
                    class="place-card__price-wrapper"
                  >
                    <div
                      class="place-card__price"
                    >
                      <b
                        class="place-card__price-value"
                      >
                        €
                        129
                      </b>
                      <span
                        class="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      class="place-card__bookmark-button
      place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        class="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlink:href="#icon-bookmark"
                        />
                      </svg>
                      <span
                        class="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    class="place-card__rating rating"
                  >
                    <div
                      class="place-card__stars rating__stars"
                    >
                      <span
                        style="width: 60%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    class="place-card__name"
                  >
                    <a
                      href="/offer/11"
                    >
                      Nice, cozy, warm big bed apartment
                    </a>
                  </h2>
                  <p
                    class="place-card__type"
                  >
                    Room
                  </p>
                </div>
              </article>
            </div>
          </section>
          <div
            class="cities__right-section"
          >
            <section
              class="cities__map map"
            >
              <div
                class="leaflet-container leaflet-grab leaflet-touch-drag"
                id="map"
                style="height: 100%; position: relative;"
                tabindex="0"
              >
                <div
                  class="leaflet-pane leaflet-map-pane"
                  style="left: 0px; top: 0px;"
                >
                  <div
                    class="leaflet-pane leaflet-tile-pane"
                  >
                    <div
                      class="leaflet-layer "
                      style="z-index: 1;"
                    >
                      <div
                        class="leaflet-tile-container leaflet-zoom-animated"
                        style="z-index: 18; left: 0px; top: 0px;"
                      >
                        <img
                          alt=""
                          class="leaflet-tile"
                          role="presentation"
                          src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                          style="width: 256px; height: 256px; left: -130px; top: -72px;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="leaflet-pane leaflet-shadow-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-overlay-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-marker-pane"
                  >
                    <img
                      alt=""
                      class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                      src="img/pin.svg"
                      style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -5px; top: 159px; z-index: 159;"
                      tabindex="0"
                    />
                  </div>
                  <div
                    class="leaflet-pane leaflet-tooltip-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-popup-pane"
                  />
                </div>
                <div
                  class="leaflet-control-container"
                >
                  <div
                    class="leaflet-top leaflet-left"
                  />
                  <div
                    class="leaflet-top leaflet-right"
                  />
                  <div
                    class="leaflet-bottom leaflet-left"
                  />
                  <div
                    class="leaflet-bottom leaflet-right"
                  >
                    <div
                      class="leaflet-control-attribution leaflet-control"
                    >
                      <a
                        href="https://leafletjs.com"
                        title="A JS library for interactive maps"
                      >
                        Leaflet
                      </a>
                       | © 
                      <a
                        href="https://www.openstreetmap.org/copyright"
                      >
                        OpenStreetMap
                      </a>
                      
          contributors © 
                      <a
                        href="https://carto.com/attributions"
                      >
                        CARTO
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Test 'MainScreen' MainScreen should render empty page if there are no offers 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
                style="display: flex;"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index page__main--index-empty"
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Paris"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Cologne"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Brussels"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Amsterdam"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Hamburg"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="/city/Dusseldorf"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container cities__places-container--empty container"
        >
          <section
            class="cities__no-places"
          >
            <div
              class="cities__status-wrapper tabs__content"
            >
              <b
                class="cities__status"
              >
                No places to stay available
              </b>
              <p
                class="cities__status-description"
              >
                We could not find any property available at the moment in undefined
              </p>
            </div>
          </section>
          <div
            class="cities__right-section"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
